GitUI/Avatars/InitialsAvatarProvider.cs

```diff
@@ -65,35 +87,76 @@ private static (string? name, char[]? separator) NameSelector(string? name, stri
                 return null;
             }
 
+            string name = names[0];
+
             // If only a single valid name-element is found ...
             if (names.Length == 1)
             {
                 // ... and that name-element is only a single character long ...
-                if (names[0].Length == 1)
+                if (name.Length == 1)
                 {
                     // ... return that character as uppercase
-                    return names[0][0].ToString().ToUpper();
+                    return name[0].ToString().ToUpper();
+                }
+
+                if (char.IsUpper(name[1]))
+                {
+                    return $"{char.ToUpper(name[0])}{name[1]}";
+                }
+
+                string[] splitNames = name.Split(_emailInitialSeparator);
+
+                if (splitNames.Length > 1)
+                {
+                    return GetInitialsFromNames(splitNames);
+                }
+
+                char[] upperChars = name.Where(char.IsUpper).ToArray();
+                if (upperChars.Length > 1)
+                {
+                    return $"{upperChars[0]}{upperChars[upperChars.Length - 1]}";
                 }
 
                 // return first letter upper-case and second letter original/lower case.
-                return $"{char.ToUpper(names[0][0])}{names[0][1]}";
+                return $"{char.ToUpper(name[0])}{name[1]}";
             }
 
             // Return initials from first and last name-element as uppercase
-            return $"{names[0][0]}{names[names.Length - 1][0]}".ToUpper();
+            return $"{name[0]}{names[names.Length - 1][0]}".ToUpper();
         }
 
         private readonly Graphics _graphics = Graphics.FromImage(new Bitmap(1, 1));
         private readonly Brush _textBrush = new SolidBrush(Color.WhiteSmoke);
 
-        private readonly Color[] _avatarColors =
+        private readonly Color[] _avatarColors = AppSettings.AvatarAuthorInitialsPalette.Split(",").Select(ConvertToColor).WhereNotNull().Cast<Color>().ToArray();
+
+        private static Color? ConvertToColor(string c)
         {
-            Color.RoyalBlue,
-            Color.DarkRed,
-            Color.Purple,
-            Color.ForestGreen,
-            Color.DarkOrange
-        };
+            if (!c.StartsWith("#"))
+            {
+                try
+                {
+                    return Color.FromName(c);
+                }
+                catch (Exception)
+                {
+                    return Color.Black;
+                }
+            }
+
+            try
+            {
+                var alpha = Convert.ToInt32(c.Substring(1, 2), 16);
+                var red = Convert.ToInt32(c.Substring(3, 2), 16);
+                var green = Convert.ToInt32(c.Substring(5, 2), 16);
+                var blue = Convert.ToInt32(c.Substring(7, 2), 16);
```

Comment:
use `int`?

Reply:
Or use the class already made for this purpose. 

https://learn.microsoft.com/en-us/dotnet/api/system.drawing.colorconverter

Reply:
Or just pass the full number instead of substring 

https://learn.microsoft.com/en-us/dotnet/api/system.drawing.color.fromargb

Reply:
https://www.c-sharpcorner.com/uploadfile/mahesh/the-colorconverter-and-colortranslater-classes-in-gdi/

Basically use the right one based on what you want to store and retrieve.

Reply:
@vbjay do you mean [`ColorTranslator.FromHtml(String)`](https://learn.microsoft.com/en-us/dotnet/api/system.drawing.colortranslator.fromhtml?view=net-7.0#system-drawing-colortranslator-fromhtml(system-string))?

Reply:
done.

Reply:
@RussKie  yes. or the color converter way... Either way we shouldn't be rolling our own way when .net has ways to translate colors.

Reply:
Once you have the color you can retrieve the parts.
```c#
Color c = Colors.Cyan;

Alpha= c.A;
Red=c.R;
```